Add IPv6 support

ton31337 opened the pull request.

ton31337 commented:
ping

ton31337 commented:
A year+ passed, someone to review? :smile:

ton31337 commented:
ping @dd32 @GaryJones

dd32 commented on readme.txt:
```diff
@@ -36,7 +36,8 @@ Add something similar to the following to wp-config.php above `/* That's all, st
 		'10.10.10.20:11211',
-		'10.10.10.30:11211'
+		'10.10.10.30:11211',
+		'[::1]:11211
```
```suggestion
		'[::1]:11211',
```

dd32 commented on object-cache.php:
```diff
@@ -1044,11 +1044,25 @@ function __construct() {
+							$port = ini_get( 'memcache.default_port' );
+						} else {
+							list ( $node, $port ) = explode( ']:', $server, 2 );
+							$node .= "]";
```
```suggestion
							$node .= ']';
```

dd32 reviewed (commented):
Reading the changes here, I don't see any issues with it, other than a few minor coding style notes.

I think it's worth noting that it also looks like a Unix socket would continue working if it's added as such, which while undocumented, technically would work.
```
$memcached_servers = array(
	'default' => array(
		'/path/to/socket:0'
	}
);
```

ton31337 committed:
Add IPv6 support

Add Hostinger we run memcached on ::1 (IPv6-only), and noticed, that it do not
work. Let's get fix it.

Signed-off-by: Donatas Abraitis <donatas.abraitis@hostinger.com>

ton31337 commented:
Thanks for a quick review, I fixed the suggested changes. Also rebased and force-pushed.
